conan/tools/intel/oneapi.py

```diff
@@ -0,0 +1,91 @@
+"""
+    Intel® oneAPI Toolkit (DPC++/C++ Compiler)
+        - Versioning: https://software.intel.com/content/www/us/en/develop/articles/oneapi-toolkit-version-to-compiler-version-mapping.html
+        - Compiler: https://software.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top.html
+"""
+import os
+import platform
+from contextlib import contextmanager
+
+from conans.client.tools.env import environment_append, env_diff
+from conans.client.tools.win import MSVS_YEAR
+from conans.errors import ConanException
+from conans.util.env_reader import get_env
+
+
+def is_using_intel_oneapi(compiler_version):
+    """Check if the Intel compiler to be used belongs to Intel oneAPI
+
+    Note: Intel oneAPI Toolkit first version is 2021.1.1
+    """
+    return int(compiler_version.split(".")[0]) > 19
+
+
+def get_intel_installation_path(out):
+    system = platform.system()
+
+    if system == "Darwin":
+        # Let's show WARNING for macOS
+        out.warn("The macOS* operating system is not supported by the "
```

Comment:
Why not exception?

Reply:
What if you are using the classic Compiler instead of the newest one? https://software.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/compiler-setup/using-the-command-line/specifying-the-location-of-compiler-components.html